packages/core/lib/context_utils.js

```diff
@@ -205,8 +205,13 @@ var contextUtils = {
   }
 };
 
-cls.createNamespace(NAMESPACE);
-logger.getLogger().debug('Starting the AWS X-Ray SDK in automatic mode (default).');
+if (process.env.AWS_XRAY_MANUAL_MODE) {
+  cls_mode = false;
+  logger.getLogger().debug('Starting the AWS X-Ray SDK in manual mode.');
+} else {
+  cls.createNamespace(NAMESPACE);
+  logger.getLogger().debug('Starting the AWS X-Ray SDK in automatic mode (default).');
+}
```

Comment:
Is it possible to refactor this change to not rely on a new env variable (`AWS_XRAY_MANUAL_MODE`)?Possibly using just the `cls_mode` variable?

Reply:
I wish, but this is done on the module load, which doesn't give a chance to the developer to change the `cls_mode` value.
To do it at a later stage will mean a breaking change. I'm willing to implement that, but I need to know that this is ok with you and how do you want to tackle that change

Reply:
Do you mind going into more detail about what the change would be in that case and why/how it would be a breaking change? Ideally, I would want to aim for a solution that fixes this issue out of the box without needing a change from the customer's end.

Reply:
Got it, thanks for the detailed information! In that case, I agree with your approach for the fix here :)